report_async/models/report_async.py

```diff
@@ -0,0 +1,181 @@
+        action = self.env.ref(self.action_id.xml_id)
+        result = action.read()[0]
+        ctx = safe_eval(result.get('context', {}))
+        ctx.update({'async_process': False})
+        result['context'] = ctx
+        return result
+
+    @api.multi
+    def run_async(self):
+        self.ensure_one()
+        if not self.allow_async:
+            raise UserError(_('Background process not allowed.'))
+        action = self.env.ref(self.action_id.xml_id)
+        result = action.read()[0]
+        ctx = safe_eval(result.get('context', {}))
+        ctx.update({'async_process': True})
+        result['context'] = ctx
+        return result
+
+    @api.multi
+    def view_files(self):
+        self.ensure_one()
+        action = self.env.ref('report_async.action_view_files')
+        result = action.read()[0]
+        result['domain'] = [('id', 'in', self.file_ids.ids)]
+        return result
+
+    @api.multi
+    def view_jobs(self):
+        self.ensure_one()
+        action = self.env.ref('queue_job.action_queue_job')
+        result = action.read()[0]
+        result['domain'] = [('id', 'in', self.job_ids.ids)]
+        result['context'] = {}
+        return result
+
+    @api.model
+    @job
+    def run_report(self, docids, data, report_id, user_id):
+        report = self.env['ir.actions.report'].browse(report_id)
+        func = REPORT_TYPES_FUNC[report.report_type]
+        # Run report
+        out_file, file_ext = getattr(report, func)(docids, data)
+        out_file = base64.b64encode(out_file)
+        out_name = '%s.%s' % (report.name, file_ext)
+        # Save report to attachment
+        user = self.env['res.users'].browse(user_id)
+        attachment = self.env['ir.attachment'].sudo().create({
+            'name': out_name,
+            'datas': out_file,
+            'datas_fname': out_name,
+            'type': 'binary',
+            'res_model': 'report.async',
+            'res_id': self.id,
+        })
+        self._cr.execute("""
+            UPDATE ir_attachment SET create_uid = %s, write_uid = %s
+            WHERE id = %s""", (self._uid, self._uid, attachment.id))
+        # Send email
+        if self.email_notify:
+            self._send_email(user, attachment)
+
+    def _send_email(self, user, attachment):
+        base_url = self.env['ir.config_parameter'].\
+            sudo().get_param('web.base.url')
+        url = "{}/web/content/ir.attachment/{}/datas/{}?download=true".\
+            format(base_url, attachment.id, attachment.name, )
+
+        odoo_bot = self.sudo().env.ref('base.partner_root')
+        email_from = odoo_bot.email
+        description = attachment.name
+        self.env['mail.mail'].create({
+            'email_from': email_from,
+            'reply_to': email_from,
+            'recipient_ids': [(6, 0, [user.partner_id.id])],
+            'subject': _("Export {} {}").format(
+                description, fields.Date.to_string(fields.Date.today())),
+            'body_html': _("""
```

Comment:
Mmmmh, maybe a mail.template ?

Reply:
Done using a template.

![image](https://user-images.githubusercontent.com/1973598/76502112-0f125680-6476-11ea-881b-f998ba1b0c5e.png)

Reply:
Add a widget in the navigation drawer to centrally notify the report print results.
How about this suggestion？

Reply:
This is great anyway